src/sage/sets/disjoint_set.pyx

```diff
@@ -397,7 +397,7 @@ cdef class DisjointSet_of_integers(DisjointSet_class):
             sage: d.__getstate__()
             [0, 1, 0, 0, 4]
         """
-        l = []
+        cdef list l = []
         cdef int i
         for i in range(self.cardinality()):
             l.append(self._nodes.parent[i])
```

Comment:
```suggestion
        cdef Py_ssize_t card = self.cardinality()
        cdef list l = [None] * card
        cdef int i
        for i in range(card):
            l[i] = self._nodes.parent[i]
```
It is faster to set the length of the list than to constantly append. (Here it is a micro-optimization.)